pkg/kube/activity.go

```diff
@@ -161,8 +163,23 @@ func GenerateBuildNumber(activities typev1.PipelineActivityInterface, pipelines
 	return build, answer, nil
 }
 
+func lazyUpdateSourceRepositoryCRD(jxClient versioned.Interface, ns string, activity *v1.PipelineActivity) error {
+	//FIXME: this relies on a bad assumption that the name of the sourcerepositories CRD entry is always equal to the repository name
+	_, err := sourcerepository.NewSourceRepositoryService(jxClient, ns).GetSourceRepository(activity.RepositoryName())
```

Comment:
You can construct the object once and store it a variable, instead of call it twice here and in the if. e.g:

```
srs := sourcerepository.NewSourceRepositoryService(jxClient, ns)
```